### ISSUE TITLE
The Grpc.Core.RpcException type and Google.Cloud.Firestore 3.0.0

### ISSUE THREAD

**Issue by lukepuplett** [googleapis/google-cloud-dotnet#9033]
Hello - I'm in the process of updating project refs of **Google.Cloud.Firestore** to 3.0.0 from 2.5.0.

https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Firestore/latest/history

Apparently this library no longer uses **Grpc.Core** and so I've removed my `using` statements `using Grpc.Core;`. This has led to compilation errors where I'm catching **RpcException**, since the type can no longer be found.

I was expecting to find this type in some other namespace, i.e. not in **Grpc.Core**. But using Command+. to fix the error, VSCode suggests I add `using Grpc.Core;` again.

If I do add this back in, when I decompile the **RpcException** it shows it's source as coming from **Grpc.Core.Api.dll**. I'm confused: I didn't think this library was used anymore.

            try
            {
                await this.FirestoreHelper.ExecuteInsertOperationAsync(op, doc);
            }
            catch (RpcException rpcEx) when (rpcEx.Message.Contains("AlreadyExist"))
            {
                throw new StorageOperationException(
                    $"Cannot insert document. A document in collection {this.CollectionGroupId} with " +
                    $"ID {doc.DocRef.Id} already exists.", rpcEx);
            }

I'm not sure what to do. Should I use `Grpc.Core.RcpException` or change it to catch a differently named exception type? And if so, will the error message strings be the same?

If not, then why is it still using Grpc.Core? Thanks.

**Comment by jskeet** [comment]
Grpc.Core.dll isn't used any more, but Grpc.Core.Api.dll is, and that contains types in the Grpc.Core namespace. (Assembly name and namespace are orthogonal concepts; they're only similar by convention.) Basically, the original Grpc.Core package was split into Grpc.Core and Grpc.Core.Api a long time ago to allow for multiple implementations - both Grpc.Core and Grpc.Net.Client implement the abstractions in the Grpc.Core.Api package.

Basically, it's fine to have the using directive for the Grpc.Core namespace. Even though it's a major version bump, we expect *most* customers to be able to upgrade without any changes to their code at all.
